src/components/fields/DataSelector.js

```diff
@@ -108,7 +110,10 @@ export class UnconnectedDataSelector extends Component {
           optionRenderer={this.context.dataSourceOptionRenderer}
           valueRenderer={this.context.dataSourceValueRenderer}
           clearable={true}
-          placeholder={this.props.placeholder}
+          placeholder={
+            this.hasData ? 'Data inlined in figure' : 'Choose data...'
```

Comment:
well, the placeholder will never show right, it this.hasData is true..
because instead there will be something in the Data selector..

but ok

Reply:
this is for the case when you're opening a plot in JupyterLab without any data sources but inlined data.